package/MDAnalysis/lib/nsgrid.pyx

```diff
@@ -407,314 +179,172 @@ cdef class NSResults(object):
-
-        # Allocate memory
-        self.beadids = <ns_int *> PyMem_Malloc(sizeof(ns_int) * self.size * self.nbeads_per_cell)  # np.empty((self.size, nbeads_max), dtype=np.int)
-        if not self.beadids:
-            raise MemoryError("Could not allocate memory for _NSGrid.beadids ({} bits requested)".format(sizeof(ns_int) * self.size * self.nbeads_per_cell))
-
-        with nogil:
-            # Second loop: fill grid
-            for i in range(ncoords):
-
-                # Add bead to grid cell
-                cellindex = self.cellids[i]
-                self.beadids[cellindex * self.nbeads_per_cell + beadcounts[cellindex]] = i
-                beadcounts[cellindex] += 1
+        # TODO: Check optimising for orthogonal cells
+        cdef int xyz[3]
+
+        self.coord2cellxyz(coord, xyz)
+        
+        return xyz[0] + xyz[1] * self.cell_offsets[1] + xyz[2] * self.cell_offsets[2]
+
+    cdef void coord2cellxyz(self, const float* coord, int* xyz) nogil:
+        """Calculate cell coordinate for coord"""
+        cdef int cx, cy, cz
+        
+        cz = <int> (coord[2] / self.cellsize[2])
+        cy = <int> ((coord[1] - cz * self.triclinic_dimensions[7]) / self.cellsize[1])
+        cx = <int> ((coord[0] - cy * self.triclinic_dimensions[3] - cz * self.triclinic_dimensions[6]) / self.cellsize[0])
+        
+        xyz[0] = cx
+        xyz[1] = cy
+        xyz[2] = cz
+
+    cdef int cellxyz2cellid(self, int cx, int cy, int cz) nogil:
+        """Convert cell coordinate to cell id"""
+        # TODO: return -1 on overflow for periodic
+        # periodic boundaries on cell coordinates
+        if cx < 0:
+            cx = self.ncells[0] - 1
+        elif cx == self.ncells[0]:
+            cx = 0
+        if cy < 0:
+            cy = self.ncells[1] - 1
+        elif cy == self.ncells[1]:
+            cy = 0
+        if cz < 0:
+            cz = self.ncells[2] - 1
+        elif cz == self.ncells[2]:
+            cz = 0
+
+        return cx + cy * self.cell_offsets[1] + cz * self.cell_offsets[2]
 
 
 cdef class FastNS(object):
     """Grid based search between two group of atoms
 
-    Instantiates a class object which uses :class:`_PBCBox` and
-    :class:`_NSGrid` to construct a cuboidal
-    grid in an orthogonal brick shaped box.
-
     Minimum image convention is used for distance evaluations
     if pbc is set to ``True``.
     """
-    cdef _PBCBox box
-    cdef real[:, ::1] coords
-    cdef real[:, ::1] coords_bbox
-    cdef readonly dreal cutoff
+    cdef float[:, ::1] coords_bbox
+    cdef readonly double cutoff
 
     cdef _NSGrid grid
-    cdef ns_int max_gridsize
     cdef bint periodic
 
-    def __init__(self, cutoff, coords, box, max_gridsize=5000, pbc=True):
+    def __init__(self, cutoff, coords, box, pbc=True):
         """
         Initialize the grid and sort the coordinates in respective
         cells by shifting the coordinates in a brick shaped box.
```

Comment:
This is no longer true, coordinates are not necessarily shifted, and the box stays as it is. Also the following two lines are now obsolete.
Also, the lines about a pseudobox don't really make sense to me here... Isn't the pseudobox automatically created if `box is None`?